src/types/core/pool.ts

```diff
@@ -241,3 +242,21 @@ export function getAssetsOrder(pool: Pool, assetInfo: AssetInfo) {
 		return [pool.assets[1], pool.assets[0]] as Array<Asset>;
 	}
 }
+
+/**
+ * Function to remove pools that are not used in paths.
+ * @param pools Array of Pool types to check for filtering.
+ * @param paths Array of Path types to check the pools against.
+ * @returns Filtered array of Pools.
+ */
+export function removedUnusedPools(pools: Array<Pool>, paths: Array<Path>): Array<Pool> {
+	const filteredPools: Array<Pool> = [];
+	pools.forEach((pool) =>
+		paths.forEach((path) => {
+			if (path.pools.find((pathPool) => pathPool.address === pool.address) !== undefined) {
+				filteredPools.push(pool);
+			}
+		}),
+	);
+	return [...new Set(filteredPools)];
```

Comment:
Probably could simplify this by using making `filteredPools` a `Set` instead of an `Array`.

Reply:
Will this error when we try to push an object thats already present? @kaimen-sano